### PR TITLE
Added a subtask: TASK_COMPILE_WARP_GET_HASH

### PR DESCRIPTION
**Pull request by Ayush-Jalan**
TASK_COMPILE_WARP_GET_HASH
Calculates and stores the hashes of the contracts and checks if they need compilation

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -13,6 +14,30 @@ export function colorLogger(str: any) {
...
+    const readData = fs.readFileSync('warp_output/hash.json', 'utf-8');
+    const existingData = JSON.parse(readData) as HashInfo[];
+    existingData.forEach((ctr) => {
+      const temp = new HashInfo('', '');
+      Object.assign(temp, ctr);
+      if (temp.getSolidityFile() === hash.getSolidityFile()) {
+        if (temp.getHash() === hash.getHash()) {
+          needToCompile = false;
+        } else {
+          temp.setHash(hash.getHash());
```
Comment (Glitch18) [inline src/utils.ts:30]: We don't need to update the hash of the temp object. The `HashInfo` object with the new hash already exists in the `hashes` array and will be written out to the JSON file at the end of this function.

File: src/Hash.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  getSolidityFile() {
+    return this.solidityFile;
+  }
+
+  getHash() {
+    return this.hash;
+  }
+
+  setHash(hash: string) {
+    this.hash= hash;
```
Comment (Glitch18) [inline src/Hash.ts:19]: A small nitpick here, but let's ensure code formatting by adding a space to either side of the `=`

**Review by Glitch18 — CHANGES_REQUESTED** [review]
Looks good overall. Some small nitpicks here and there, but apart from that it should get the feature working. 

Also, please delete the .DS_Store, package-lock.json and yarn.lock files.

File: src/index.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import './type-extensions';
-
+import * as fs from 'fs';
+const {createHash} = require('crypto');
```
[the remark this answers is not in the document]
Comment (Glitch18) [reply src/index.ts:3]: If not, you can ignore this

[the remark this answers is not in the document]
Comment (Ayush-Jalan) [reply src/index.ts:3]: It is available in Node.js v18.8.0, so I don't think we need to add it

**Comment by Ayush-Jalan** [comment]
I have made the changes and deleted the files, you can have a look.

**Review by Glitch18 — APPROVED** [review]

**Pull request merged**